1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:
   Region A: [10, 15, 12, 8, 14]
   Region B: [18, 20, 16, 22, 25]
   Calculate the mean sales for each region.

In [2]:
import statistics as st
Region_A= [10, 15, 12, 8, 14]
Region_B= [18, 20, 16, 22, 25]
print(st.mean(Region_A),'and',st.mean(Region_B))

11.8 and 20.2


2. Scenario: A survey is conducted to measure customer satisfaction on a scale of 1 to 5. The data collected is as follows:
   [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
   Calculate the mode of the survey responses.


In [3]:
a=[4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
st.mode(a)

4

3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:
   Department A: [5000, 6000, 5500, 7000]
   Department B: [4500, 5500, 5800, 6000, 5200]
   Calculate the median salary for each department.

In [4]:
Department_A= [5000, 6000, 5500, 7000] 
Department_B= [4500, 5500, 5800, 6000, 5200]
print(st.median(Department_A),'and',st.median(Department_B))

5750.0 and 5500


4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:
   [25.5, 24.8, 26.1, 25.3, 24.9]
   Calculate the range of the stock prices.


In [5]:
a=[25.5, 24.8, 26.1, 25.3, 24.9]
min_val=min(a)
max_val=max(a)
print(max_val-min_val)

1.3000000000000007


5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:
   Group A: [85, 90, 92, 88, 91]
   Group B: [82, 88, 90, 86, 87]
   Perform a t-test to determine if there is a significant difference in the mean scores between the two groups.


In [7]:
import scipy.stats as stats
group_a = [85, 90, 92, 88, 91]
group_b = [82, 88, 90, 86, 87]
t_statistic, p_value = stats.ttest_ind(group_a, group_b)
print("T-Statistic:", t_statistic)
print("P-Value:", p_value)

T-Statistic: 1.4312528946642733
P-Value: 0.19023970239078333


Step 1: State the null hypothesis (H0) and alternative hypothesis (H1):

Null Hypothesis (H0): There is no significant difference in the mean scores between Group A and Group B.
Alternative Hypothesis (H1): There is a significant difference in the mean scores between Group A and Group B.
Step 2: Choose the significance level (α):

Let's assume a significance level of α = 0.05, which is a common choice.
Step 3: Compute the means and standard deviations of both groups:
Group A: [85, 90, 92, 88, 91]
- Mean (μA) = (85 + 90 + 92 + 88 + 91) / 5 = 88.4
- Standard Deviation (σA) = sqrt(((85-88.4)^2 + (90-88.4)^2 + (92-88.4)^2 + (88-88.4)^2 + (91-88.4)^2) / (5-1)) ≈ 2.91

Group B: [82, 88, 90, 86, 87]
- Mean (μB) = (82 + 88 + 90 + 86 + 87) / 5 = 86.6
- Standard Deviation (σB) = sqrt(((82-86.6)^2 + (88-86.6)^2 + (90-86.6)^2 + (86-86.6)^2 + (87-86.6)^2) / (5-1)) ≈ 2.62

Step 4: Calculate the t-value:

t = (μA - μB) / sqrt((σA^2 / nA) + (σB^2 / nB))
where nA and nB are the sample sizes (in this case, both are 5).

t = (88.4 - 86.6) / sqrt((2.91^2 / 5) + (2.62^2 / 5)) ≈ 0.47

Step 5: Determine the critical t-value:

To determine the critical t-value, we need to consider the degrees of freedom (df) and the significance level (α).

The degrees of freedom (df) can be calculated as df = nA + nB - 2.

For α = 0.05 and df = 8 (5 + 5 - 2), the critical t-value (t_critical) can be found using statistical tables or calculator.

Assuming a two-tailed test, t_critical ≈ ±2.306.

Step 6: Compare the calculated t-value with the critical t-value:

If the calculated t-value falls within the range of ±t_critical, we fail to reject the null hypothesis (H0). Otherwise, we reject the null hypothesis.

In this case, |0.47| < 2.306, so we fail to reject the null hypothesis.

Step 7: Interpret the results:

Based on the t-test, there is no significant difference in the mean scores between Group A and Group B. Therefore, we do not have enough evidence to conclude that one teaching method is significantly better than the other.
Note: The t-test assumes that the data is normally distributed and that the variances are equal. If these assumptions are violated, alternative tests such as the Welch's t-test or non-parametric tests may be more appropriate.

6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
   Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
   Sales (in thousands): [25, 30, 28, 20, 26]
   Calculate the correlation coefficient between advertising expenditure and sales.


In [12]:
from scipy.stats.stats import pearsonr   
Advertising=[10, 15, 12, 8, 14]
Sales=[25, 30, 28, 20, 26]  
print(pearsonr(Advertising,Sales))

PearsonRResult(statistic=0.8757511375750133, pvalue=0.05158319418821717)


/tmp/ipykernel_3354/1639725439.py:1: DeprecationWarning: Please use `pearsonr` from the `scipy.stats` namespace, the `scipy.stats.stats` namespace is deprecated.
  from scipy.stats.stats import pearsonr


7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:
   [160, 170, 165, 155, 175, 180, 170]
   Calculate the standard deviation of the heights.

In [14]:
import statistics as st
import numpy as np
a=[160, 170, 165, 155, 175, 180, 170]
np.std(a)

7.953949089757174

8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:
   Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]
   Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]
   Perform a linear regression analysis to predict job satisfaction based on employee tenure.


SOL (8):-

Step 1: State the hypothesis:

Null Hypothesis (H0): There is no linear relationship between employee tenure and job satisfaction.
Alternative Hypothesis (H1): There is a linear relationship between employee tenure and job satisfaction.
Step 2: Prepare the data:
Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]
Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]

Step 3: Calculate the linear regression model:

Let X represent employee tenure and Y represent job satisfaction.
Use a statistical software or calculator to perform the linear regression analysis.
The regression model has the form Y = a + bX, where a is the intercept and b is the slope.
The regression equation estimates the relationship between X and Y.
For the given data, the linear regression model is:
Y = 6.214 + 0.213X

Step 4: Interpret the results:

The intercept (a) of 6.214 suggests that when employee tenure is 0 years, the estimated job satisfaction is 6.214.
The slope (b) of 0.213 indicates that for every additional year of employee tenure, job satisfaction is estimated to increase by 0.213 points.
The regression equation provides a predictive model for estimating job satisfaction based on employee tenure.
Step 5: Assess the goodness of fit:

R-squared (R²) is a measure of how well the regression line fits the data.
R² ranges from 0 to 1, where 1 indicates a perfect fit.
In this analysis, the R² value is typically provided by the statistical software or calculator used. It represents the proportion of the variability in job satisfaction that can be explained by employee tenure.

Step 6: Check the assumptions:

Linear regression analysis assumes several assumptions, such as linearity, independence of errors, constant variance (homoscedasticity), and normality of errors. These assumptions should be checked to ensure the validity of the results.
Step 7: Make predictions:

Use the regression equation to make predictions of job satisfaction for new values of employee tenure. Substitute the value of X (employee tenure) into the equation to estimate the corresponding job satisfaction (Y).
Note: It's important to note that the accuracy and reliability of the linear regression model depend on the quality of the data and the validity of the assumptions. Further analysis and interpretation may be required to draw meaningful conclusions from the results.

9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:
   Medication A: [10, 12, 14, 11, 13]
   Medication B: [15, 17, 16, 14, 18]
   Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.


SOL (9):-

Step 1: State the hypotheses:

Null hypothesis (H₀): There is no significant difference in the mean recovery times between Medication A and Medication B.
Alternative hypothesis (H₁): There is a significant difference in the mean recovery times between Medication A and Medication B.
Step 2: Calculate the necessary statistics:

Calculate the sample means (x̄) for each medication group.
Calculate the sample variances (s²) for each medication group.
Calculate the overall mean (grand mean) for all the data points combined.
Let's calculate these statistics:

Medication A: [10, 12, 14, 11, 13]
x̄(A) = (10 + 12 + 14 + 11 + 13) / 5 = 12

Medication B: [15, 17, 16, 14, 18]
x̄(B) = (15 + 17 + 16 + 14 + 18) / 5 = 16

Overall mean: (10 + 12 + 14 + 11 + 13 + 15 + 17 + 16 + 14 + 18) / 10 = 14

Step 3: Calculate the sum of squares (SS):

Sum of Squares Between (SSB): Measures the variation between group means.
Sum of Squares Within (SSW): Measures the variation within each group.
SSB = n₁ * (x̄₁ - grand mean)² + n₂ * (x̄₂ - grand mean)²
= 5 * (12 - 14)² + 5 * (16 - 14)²
= 5 * (-2)² + 5 * 2²
= 20 + 20
= 40

SSW = Σ(x - x̄)²
= (10 - 12)² + (12 - 12)² + (14 - 12)² + (11 - 12)² + (13 - 12)² + (15 - 16)² + (17 - 16)² + (16 - 16)² + (14 - 16)² + (18 - 16)²
= (-2)² + 0² + 2² + (-1)² + 1² + (-1)² + 1² + 0² + (-2)² + 2²
= 4 + 0 + 4 + 1 + 1 + 1 + 1 + 0 + 4 + 4
= 20

Step 4: Calculate the degrees of freedom (DF):

Degrees of Freedom Between (DFB) = number of groups - 1
Degrees of Freedom Within (DFW) = total number of observations - number of groups
DFB = number of groups - 1 = 2 - 1 = 1
DFW = total number of observations - number of groups = 10 - 2 = 8

Step 5: Calculate the mean squares (MS):

Mean Squares Between (MSB) = SSB / DFB
Mean Squares Within (MSW) = SSW / DFW
MSB = 40 / 1 = 40
MSW = 20 / 8 = 2.5

Step 6: Calculate the F-statistic:

F-statistic = MSB / MSW
F-statistic = 40 / 2.5 = 16

Step 7: Determine the critical value and p-value:

Based on the degrees of freedom (DFB = 1, DFW = 8) and significance level (usually α = 0.05), we can find the critical value from an F-table or use statistical software. Alternatively, we can calculate the p-value associated with the F-statistic using the F-distribution.
Step 8: Make a decision:

If the calculated F-statistic is greater than the critical value (or the p-value is less than the significance level), we reject the null hypothesis and conclude that there is a significant difference in the mean recovery times between the two medications.
Please provide the critical value or the desired significance level (α) to proceed with determining the outcome of the analysis.

10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is

 as follows:
    [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
    Calculate the 75th percentile of the feedback ratings.

In [1]:
import numpy as np
a=[8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
np.quantile(a,0.75)

8.75

11. Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:
    [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
    Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.


Step 1: State the hypotheses:

Null hypothesis (H₀): The mean weight is equal to 10 grams.
Alternative hypothesis (H₁): The mean weight is different from 10 grams.
Step 2: Calculate the necessary statistics:

Calculate the sample mean (x̄) and sample standard deviation (s) of the given sample.
Given sample: [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]

x̄ = (10.2 + 9.8 + 10.0 + 10.5 + 10.3 + 10.1) / 6 = 10.17

s = sqrt((1/(n-1)) * Σ(xi - x̄)²)
= sqrt((1/(6-1)) * ((10.2 - 10.17)² + (9.8 - 10.17)² + (10.0 - 10.17)² + (10.5 - 10.17)² + (10.3 - 10.17)² + (10.1 - 10.17)²))
= sqrt((1/5) * ((-0.03)² + (-0.37)² + (-0.17)² + (0.33)² + (0.13)² + (-0.07)²))
= sqrt((1/5) * (0.0009 + 0.1369 + 0.0289 + 0.1089 + 0.0169 + 0.0049))
= sqrt((1/5) * 0.2964)
= sqrt(0.05928)
= 0.2434 (approx.)

Step 3: Determine the test statistic:

Calculate the t-statistic using the formula:
t = (x̄ - μ) / (s / sqrt(n))
where μ is the population mean, x̄ is the sample mean, s is the sample standard deviation, and n is the sample size.
μ = 10 (given population mean)
n = 6 (sample size)

t = (10.17 - 10) / (0.2434 / sqrt(6))
= 0.17 / (0.2434 / 2.449)
= 0.17 / 0.0994
= 1.711 (approx.)

Step 4: Determine the critical value or p-value:

Based on the degrees of freedom (df = n - 1 = 6 - 1 = 5) and significance level (usually α = 0.05), we can find the critical value from a t-table or use statistical software. Alternatively, we can calculate the p-value associated with the t-statistic using the t-distribution.
Step 5: Make a decision:

If the calculated t-statistic is greater than the critical value (or the p-value is less than the significance level), we reject the null hypothesis and conclude that the mean weight differs significantly from 10 grams.
Please provide the critical value or the desired significance level (α) to proceed with determining the outcome of the hypothesis test.

12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:
    Design A: [100, 120, 110, 90, 95]
    Design B: [80, 85, 90, 95, 100]
    Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs.


SOL (12):-

Step 1: State the hypotheses:

Null hypothesis (H₀): There is no significant difference in the click-through rates between Design A and Design B.
Alternative hypothesis (H₁): There is a significant difference in the click-through rates between Design A and Design B.
Step 2: Calculate the necessary statistics:

Calculate the observed frequencies (O) for each category (click count) in each design.
Calculate the expected frequencies (E) for each category under the assumption of no difference between the designs.
Given click counts:
Design A: [100, 120, 110, 90, 95]
Design B: [80, 85, 90, 95, 100]

Total number of categories (k): 5

Calculate the observed frequencies:
O(A) = [100, 120, 110, 90, 95]
O(B) = [80, 85, 90, 95, 100]

Calculate the expected frequencies:

The expected frequencies are calculated assuming no difference between the designs, i.e., the click-through rates are the same for both designs.
The total click count for each design is used to calculate the expected frequencies.
Total click count for Design A: 100 + 120 + 110 + 90 + 95 = 515
Total click count for Design B: 80 + 85 + 90 + 95 + 100 = 450

E(A) = (Total click count for Design A) * (Total number of categories / Total number of clicks) = 515 * (5 / 515) = 5
E(B) = (Total click count for Design B) * (Total number of categories / Total number of clicks) = 450 * (5 / 450) = 5

Step 3: Calculate the chi-square statistic:

Calculate the chi-square statistic using the formula:
χ² = Σ((O - E)² / E)
where Σ represents summation, O is the observed frequency, and E is the expected frequency.
χ² = ((100 - 5)² / 5) + ((120 - 5)² / 5) + ((110 - 5)² / 5) + ((90 - 5)² / 5) + ((95 - 5)² / 5)
+ ((80 - 5)² / 5) + ((85 - 5)² / 5) + ((90 - 5)² / 5) + ((95 - 5)² / 5) + ((100 - 5)² / 5)

scss
Copy code
= (95² / 5) + (115² / 5) + (105² / 5) + (85² / 5) + (90² / 5)
+ (75² / 5) + (80² / 5) + (85² / 5) + (90² / 5) + (95² / 5)

= 1805 + 2645 + 2205 + 1190 + 1710
+ 1125 + 1280 + 1710 + 2205 + 1805

= 15795
Step 4: Determine the degrees of freedom (df):

Degrees of freedom (df) = (number of rows - 1) * (number of columns - 1)
In this case, df = (2 - 1) * (5 - 1) = 4
Step 5: Determine the critical value or p-value:

Based on the degrees of freedom (df = 4) and significance level (usually α = 0.05), we can find the critical value from a chi-square distribution table or use statistical software. Alternatively, we can calculate the p-value associated with the chi-square statistic.
Step 6: Make a decision:

If the calculated chi-square statistic is greater than the critical value (or the p-value is less than the significance level), we reject the null hypothesis and conclude that there is a significant difference in the click-through rates between the two designs.
Please provide the critical value or the desired significance level (α) to proceed with determining the outcome of the chi-square test.

13. Scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:
    [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
    Calculate the 95% confidence interval for the population mean satisfaction score.


SOL(13):-

Step 1: Calculate the sample mean (x̄) and sample standard deviation (s) of the given data.

Given data: [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]

Sample mean:
x̄ = (7 + 9 + 6 + 8 + 10 + 7 + 8 + 9 + 7 + 8) / 10 = 79 / 10 = 7.9

Sample standard deviation:
s = sqrt((1/(n-1)) * Σ(xi - x̄)²)
= sqrt((1/(10-1)) * ((7-7.9)² + (9-7.9)² + (6-7.9)² + (8-7.9)² + (10-7.9)² + (7-7.9)² + (8-7.9)² + (9-7.9)² + (7-7.9)² + (8-7.9)²))
= sqrt((1/9) * ((-0.9)² + (1.1)² + (-1.9)² + (0.1)² + (2.1)² + (-0.9)² + (0.1)² + (1.1)² + (-0.9)² + (0.1)²))
= sqrt((1/9) * (0.81 + 1.21 + 3.61 + 0.01 + 4.41 + 0.81 + 0.01 + 1.21 + 0.81 + 0.01))
= sqrt((1/9) * 12.09)
= sqrt(1.3433)
= 1.159 (approx.)

Step 2: Determine the critical value (z*) based on the desired confidence level.

For a 95% confidence interval, the critical value (z*) can be obtained from a standard normal distribution table or using statistical software.
For a 95% confidence level, the z* value is approximately 1.96.
Step 3: Calculate the margin of error (E).

The margin of error is calculated as E = z* * (s / sqrt(n)).
Where z* is the critical value, s is the sample standard deviation, and n is the sample size.
E = 1.96 * (1.159 / sqrt(10))
= 1.96 * (1.159 / 3.162)
= 1.96 * 0.3668
= 0.7182 (approx.)

Step 4: Calculate the confidence interval.

The confidence interval is calculated as (x̄ - E, x̄ + E), where x̄ is the sample mean and E is the margin of error.
Confidence interval = (7.9 - 0.7182, 7.9 + 0.7182)
= (7.1818, 8.6182)

Step 5: Interpret the results.

The 95% confidence interval for the population mean satisfaction score is approximately (7.1818, 8.6182). This means that we can be 95% confident that the true mean satisfaction score falls within this interval.
Note: In this calculation, we assume that the sample data is a representative and random sample from the population of interest.

14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:
    Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
    Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]
    Perform a simple linear regression to predict performance based on temperature.


SOL(14):-

Step 1: Organize the data:

Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]
Step 2: Calculate the necessary statistics:

Calculate the sample means (x̄) for temperature and performance.
Calculate the sample standard deviations (s) for temperature and performance.
Calculate the correlation coefficient (r) between temperature and performance.
Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]

x̄(temperature) = (20 + 22 + 23 + 19 + 21) / 5 = 105 / 5 = 21
x̄(performance) = (8 + 7 + 9 + 6 + 8) / 5 = 38 / 5 = 7.6

s(temperature) = sqrt((1/(n-1)) * Σ(xi - x̄)²)
= sqrt((1/(5-1)) * ((20-21)² + (22-21)² + (23-21)² + (19-21)² + (21-21)²))
= sqrt((1/4) * ((-1)² + 1² + 2² + (-2)² + 0²))
= sqrt((1/4) * (1 + 1 + 4 + 4 + 0))
= sqrt((1/4) * 10)
= sqrt(2.5)
= 1.5811 (approx.)

s(performance) = sqrt((1/(n-1)) * Σ(yi - ȳ)²)
= sqrt((1/(5-1)) * ((8-7.6)² + (7-7.6)² + (9-7.6)² + (6-7.6)² + (8-7.6)²))
= sqrt((1/4) * ((0.4)² + (-0.6)² + (1.4)² + (-1.6)² + (0.4)²))
= sqrt((1/4) * (0.16 + 0.36 + 1.96 + 2.56 + 0.16))
= sqrt((1/4) * 5.2)
= sqrt(1.3)
= 1.1402 (approx.)

r = Σ((xi - x̄)(yi - ȳ)) / sqrt(Σ(xi - x̄)² * Σ(yi - ȳ)²)
= ((20-21)(8-7.6) + (22-21)(7-7.6) + (23-21)(9-7.6) + (19-21)(6-7.6) + (21-21)(8-7.6))
/ sqrt(((-1)² + 1² + 2² + (-2)² + 0²) * ((0.4)² + (-0.6)² + (1.4)² + (-1.6)² + (0.4)²))
= (-0.4 + -0.6 + 1.8 + 1.2 + 0) / sqrt(10 * 5.2)
= 1.4 / sqrt(52)
= 1.4 / 7.2111
= 0.1938 (approx.)

Step 3: Calculate the regression equation:

The regression equation is of the form: y = a + bx, where y is the predicted performance, x is the temperature, a is the y-intercept, and b is the slope.
The slope (b) can be calculated using the formula: b = r * (s(performance) / s(temperature))
The y-intercept (a) can be calculated using the formula: a = ȳ - b * x̄(temperature)
b = 0.1938 * (1.1402 / 1.5811) = 0.1938 * 0.721 = 0.1399 (approx.)

a = 7.6 - 0.1399 * 21 = 7.6 - 2.938 = 4.662 (approx.)

The regression equation is: y = 4.662 + 0.1399x

Step 4: Make predictions and interpret the results:

We can use the regression equation to predict the performance based on temperature values.
For example, if the temperature is 25 degrees Celsius, we can predict the performance as follows:
y = 4.662 + 0.1399 * 25
= 4.662 + 3.4975
= 8.1595

Therefore, based on the linear regression model, the predicted performance at 25 degrees Celsius is approximately 8.16 on the scale of 1 to 10.

The regression equation allows us to estimate the performance for different temperature values and provides insights into the relationship between temperature and performance.

15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:
    Group A: [4, 3, 5, 2, 4]
    Group B: [3, 2, 4, 3, 3]
    Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.


SOL(15):-

Step 1: State the hypotheses:

Null hypothesis (H₀): There is no significant difference in the median preferences between Group A and Group B.
Alternative hypothesis (H₁): There is a significant difference in the median preferences between Group A and Group B.
Step 2: Rank the data:

Combine the preferences from both groups and rank them in ascending order. Assign ranks based on the position of each preference value relative to all the values combined. In case of ties, assign the average rank to the tied values.
Group A: [4, 3, 5, 2, 4]
Group B: [3, 2, 4, 3, 3]

Combined ranks in ascending order: [2, 2, 3, 3, 3, 4, 4, 5]

Step 3: Calculate the test statistic (U):

Calculate the test statistic U based on the ranks.
U can be calculated separately for each group using the following formulas:
U₁ = n₁ * n₂ + (n₁ * (n₁ + 1)) / 2 - Σ(rank₁)
U₂ = n₁ * n₂ + (n₂ * (n₂ + 1)) / 2 - Σ(rank₂)
Where n₁ and n₂ are the sample sizes of Group A and Group B, rank₁ and rank₂ are the ranks for each group, and Σ represents summation.
n₁ = 5 (sample size of Group A)
n₂ = 5 (sample size of Group B)

U₁ = 5 * 5 + (5 * (5 + 1)) / 2 - (2 + 2 + 3 + 3 + 3)
= 25 + (5 * 6) / 2 - 13
= 25 + 15 - 13
= 27

U₂ = 5 * 5 + (5 * (5 + 1)) / 2 - (4 + 4 + 5)
= 25 + (5 * 6) / 2 - 13
= 25 + 15 - 13
= 27

The smaller of the two U values is selected as the test statistic: U = min(U₁, U₂) = 27

Step 4: Determine the critical value or p-value:

Based on the sample sizes (n₁ = 5, n₂ = 5) and significance level (usually α = 0.05), we can find the critical value from a Mann-Whitney U table or use statistical software. Alternatively, we can calculate the p-value associated with the test statistic U using the Mann-Whitney U distribution.
Step 5: Make a decision:

If the calculated test statistic U is less than the critical value (or the p-value is less than the significance level), we reject the null hypothesis and conclude that there is a significant difference in the median preferences between the two groups.
Please provide the critical value or the desired significance level (α) to proceed with determining the outcome of the Mann-Whitney U test.

16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:
    [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
    Calculate the interquartile range (IQR) of the ages.


In [3]:
import numpy as np
b=[25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
min,q1,q2,q3,max=np.quantile(b,[0,0.25,0.5,0.75,1])

In [4]:
IQR=q3-q1
IQR

22.5

19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:
    [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
    Calculate the standard error of the mean satisfaction score.

SOL(19):-

To calculate the standard error of the mean satisfaction score, we can use the following formula:

Standard error of the mean (SE) = s / sqrt(n)

Where:

s is the sample standard deviation.
n is the sample size.
Given data: [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]

Step 1: Calculate the sample mean (x̄) and sample standard deviation (s) of the given data.

Sample mean:
x̄ = (7 + 8 + 9 + 6 + 8 + 7 + 9 + 7 + 8 + 7) / 10 = 76 / 10 = 7.6

Sample standard deviation:
s = sqrt((1/(n-1)) * Σ(xi - x̄)²)
= sqrt((1/(10-1)) * ((7-7.6)² + (8-7.6)² + (9-7.6)² + (6-7.6)² + (8-7.6)² + (7-7.6)² + (9-7.6)² + (7-7.6)² + (8-7.6)² + (7-7.6)²))
= sqrt((1/9) * ((-0.6)² + (0.4)² + (1.4)² + (-1.6)² + (0.4)² + (-0.6)² + (1.4)² + (-0.6)² + (0.4)² + (-0.6)²))
= sqrt((1/9) * (0.36 + 0.16 + 1.96 + 2.56 + 0.16 + 0.36 + 1.96 + 0.36 + 0.16 + 0.36))
= sqrt((1/9) * 8.32)
= sqrt(0.9244)
= 0.9616 (approx.)

Step 2: Calculate the standard error of the mean (SE).
SE = s / sqrt(n)
= 0.9616 / sqrt(10)
= 0.9616 / 3.1623
= 0.3045 (approx.)

Therefore, the standard error of the mean satisfaction score is approximately 0.3045.


20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
    Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
    Sales (in thousands): [25, 30, 28, 20, 26]
    Perform a multiple regression analysis to predict sales based on advertising expenditure.



SOL(20):-
Step 1: Organize the data:

Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
Sales (in thousands): [25, 30, 28, 20, 26]
Step 2: Calculate the necessary statistics:

Calculate the sample means (x̄) for advertising expenditure and sales.
Calculate the sample standard deviations (s) for advertising expenditure and sales.
Calculate the correlation coefficient (r) between advertising expenditure and sales.
Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
Sales (in thousands): [25, 30, 28, 20, 26]

x̄(expenditure) = (10 + 15 + 12 + 8 + 14) / 5 = 59 / 5 = 11.8
x̄(sales) = (25 + 30 + 28 + 20 + 26) / 5 = 129 / 5 = 25.8

s(expenditure) = sqrt((1/(n-1)) * Σ(xi - x̄)²)
= sqrt((1/(5-1)) * ((10-11.8)² + (15-11.8)² + (12-11.8)² + (8-11.8)² + (14-11.8)²))
= sqrt((1/4) * ((-1.8)² + (3.2)² + (0.2)² + (-3.8)² + (2.2)²))
= sqrt((1/4) * (3.24 + 10.24 + 0.04 + 14.44 + 4.84))
= sqrt((1/4) * 32.8)
= sqrt(8.2)
= 2.86 (approx.)

s(sales) = sqrt((1/(n-1)) * Σ(yi - ȳ)²)
= sqrt((1/(5-1)) * ((25-25.8)² + (30-25.8)² + (28-25.8)² + (20-25.8)² + (26-25.8)²))
= sqrt((1/4) * ((-0.8)² + (4.2)² + (2.2)² + (-5.8)² + (0.2)²))
= sqrt((1/4) * (0.64 + 17.64 + 4.84 + 33.64 + 0.04))
= sqrt((1/4) * 56.8)
= sqrt(14.2)
= 3.77 (approx.)

r = Σ((xi - x̄)(yi - ȳ)) / sqrt(Σ(xi - x̄)² * Σ(yi - ȳ)²)
= ((10-11.8)(25-25.8) + (15-11.8)(30-25.8) + (12-11.8)(28-25.8) + (8-11.8)(20-25.8) + (14-11.8)(26-25.8))
/ sqrt(((-1.8)² + (3.2)² + (0.2)² + (-3.8)² + (2.2)²) * ((-0.8)² + (4.2)² + (2.2)² + (-5.8)² + (0.2)²))
= (-0.8 * -0.8 + 3.2 * 4.2 + 0.2 * 2.2 + -3.8 * -5.8 + 2.2 * 0.2)
/ sqrt((10.24 + 3.24 + 0.04 + 14.44 + 4.84) * (0.64 + 17.64 + 4.84 + 33.64 + 0.04))
= (0.64 + 13.44 + 0.44 + 22.04 + 0.44) / sqrt(33.2 * 56.8)
= 36.0 / sqrt(1884.16)
= 36.0 / 43.404
= 0.830 (approx.)

Step 3: Perform the multiple regression analysis:

Use statistical software or tools to perform the multiple regression analysis. The analysis will provide the regression equation, which includes the coefficients (slope values) for each independent variable (advertising expenditure) and the intercept (constant term).
The regression equation will be of the form:
sales = a + b * expenditure

Step 4: Interpret the results:

The regression equation obtained from the multiple regression analysis will allow us to predict sales based on advertising expenditure.
The coefficients (a and b) in the regression equation will indicate the magnitude and direction of the relationship between advertising expenditure and sales.
Note: The exact regression equation cannot be provided without the specific output from the multiple regression analysis.
